src/components/Auth/ValidatedInput.tsx

```diff
@@ -0,0 +1,86 @@
+import React from "react";
+import { TextInput, Text, View } from "react-native";
+import styled from "styled-components";
+import _ from "lodash";
+
+interface IProps {
+  value: string;
+  rule: (value) => { result: boolean; errorText: string };
+  placeholder: string;
+  onChangeText: (value) => void;
+}
+interface IState {
+  error: string;
+  valid: boolean;
+}
+interface ICust {
+  debouncedValidation: () => void;
+}
+class ValidatedInput extends React.Component<IProps, IState, ICust> {
+  private debouncedValidation = _.debounce(
```

Comment:
probably ICust is not needed
private debouncedValidation : () => void = _.debounce(